### ISSUE TITLE
Missing labels in in _prepare_non_packed_dataloader?

### ISSUE THREAD

**Issue by benjaminhuo** [huggingface/trl#969]
I can see that the [_prepare_non_packed_dataloader](https://github.com/huggingface/trl/blob/main/trl/trainer/sft_trainer.py#L367
) only returns `input_ids` and `attention_mask` without `labels`
```python
            return {"input_ids": outputs["input_ids"], "attention_mask": outputs["attention_mask"]}
```

But in [ConstantLengthDataset](https://github.com/huggingface/trl/blob/main/trl/trainer/utils.py#L589), the `labels` is added together with `input_ids`, `attention_mask` is not added in this case which is confusing.
```python
            for example in examples:
                self.current_size += 1
                yield {
                    "input_ids": torch.LongTensor(example),
                    "labels": torch.LongTensor(example),
                }
```

_prepare_non_packed_dataloader and ConstantLengthDataset are [different branches](https://github.com/huggingface/trl/blob/main/trl/trainer/sft_trainer.py#L315) to handle packing and non-packing dataset, in my understanding `input_ids`, `attention_mask`, and `labels` should be returned together for a dataset?

**Comment by lvwerra** [comment]
If no `labels` are passed to a causal LM then the input_ids are converted to labels inside the model. So although this is not quite consistent the effect should be the same.
